.github/workflows/go-check.yaml

```diff
@@ -48,6 +48,11 @@ jobs:
         run: |
           cd src/github.com/cilium/cilium
           make precheck
+      - name: Go code nilness checks
+        run: |
+          cd src/github.com/cilium/cilium
+          go get golang.org/x/tools/go/analysis/passes/nilness/cmd/nilness
+          make nilness
```

Comment:
Currently failing due to:

```
! (set +o pipefail; \
	    GO111MODULE=off go vet -vettool= ./api/... ./bugtool/... ./cilium/... ./cilium-health/... ./daemon/... ./hubble-relay/... ./operator/... ./pkg/... ./plugins/... ./proxylib/... ./test/. ./test/config/... ./test/ginkgo-ext/... ./test/helpers/... ./test/runtime/... ./test/k8sT/... ./tools/... 2>&1 \
	    | grep -Ev '(^#)|(^/tmp)|(vendor)|(zz_generated)' \
	    | grep '.*')
Error: api/v1/client/cilium_api_client.go:16:2: cannot find package "github.com/cilium/cilium/api/v1/client/daemon" in any of:
	/opt/hostedtoolcache/go/1.15.5/x64/src/github.com/cilium/cilium/api/v1/client/daemon (from $GOROOT)
	/home/runner/go/src/github.com/cilium/cilium/api/v1/client/daemon (from $GOPATH)
```

I don't suppose @sayboras has any thoughts / tips on how to help go vet find these paths?

Reply:
I am thinking about something like below for hipness check, what do you think ?

```diff
vagrant@runtime1:~/go/src/github.com/cilium/cilium$ git diff
diff --git a/.golangci.yaml b/.golangci.yaml
index a9db75420..e3048d8f8 100644
--- a/.golangci.yaml
+++ b/.golangci.yaml
@@ -48,6 +48,9 @@ run:

 # all available settings of specific linters
 linters-settings:
+  govet:
+    enable:
+      - nilness
   misspell:
     # Correct spellings using locale preferences for US or UK.
     # Default is to use a neutral variety of English.
```

```shell script
vagrant@runtime1:~/go/src/github.com/cilium/cilium$ make lint
golangci-lint run
pkg/k8s/version/version.go:278:9: nilness: impossible condition: nil != nil (govet)
	if err != nil {
	       ^
pkg/crypto/certificatemanager/certificate_manager.go:81:33: nilness: impossible condition: nil != nil (govet)
		if len(secrets) == 0 && ioErr != nil {
		                              ^
Makefile:472: recipe for target 'lint' failed
make: *** [lint] Error 1
```

Reply:
Thanks, great suggestion, no need for the makefile changes.